x-pack/platform/plugins/private/gen_ai_settings/public/contexts/enabled_features_context.tsx

```diff
@@ -52,14 +52,24 @@ export const EnabledFeaturesContextProvider: FC<PropsWithChildren<Props>> = ({
     const showAiAssistantsVisibilitySetting =
       config.showAiAssistantsVisibilitySetting === false ? false : !isSolutionView;
 
+    const hasObservabilityAssistant =
+      services.application.capabilities.observabilityAIAssistant?.show === true;
```

Comment:
nit, `capabilities` could be extracted on line 40. just to make this code a bit more readable.
```ts
  const {
    services: {
      spaces,
      application: { capabilities },
    },
  } = useKibana();
```